src/TestSuite/Fixture/FixtureHelper.php

```diff
@@ -159,11 +171,71 @@ public function insert(array $fixtures): void
     public function truncate(array $fixtures): void
     {
         $this->runPerConnection(function (ConnectionInterface $connection, array $groupFixtures) {
-            $connection->disableConstraints(function () use ($connection, $groupFixtures) {
-                foreach ($groupFixtures as $fixture) {
+            if ($connection instanceof Connection) {
+                $sortedFixtures = $this->sortByConstraint($connection, $groupFixtures);
+            }
+
+            $driver = $connection->getDriver();
+            if (
+                isset($sortedFixtures) &&
+                (
+                    $driver instanceof Postgres ||
+                    $driver instanceof Sqlite ||
+                    $driver instanceof Sqlserver
+                )
+            ) {
+                foreach (array_reverse($sortedFixtures) as $fixture) {
                     $fixture->truncate($connection);
                 }
-            });
+            } else {
+                $connection->disableConstraints(function () use ($connection, $groupFixtures) {
```

Comment:
I wonder if we should have a wrapper on the `Connection`/`Driver` that can encapsulate these platform differences better. I'm thinking a method like 

```php
$connection->prepareTruncate(function ($connection) use ($groupFixtures) {
  foreach ($groupFixtures as $fixture) {
    $fixture->truncate($connection); 
 }
});
```

Could work. The naming might be awkward, but it lets us encapsulate the driver specific details into a method with a similar name, and keep most of the logic in the driver. This could be helpful if oracle or a future driver needs different semantics around transactions & constraints being disabled.

Reply:
Do we need to fix it for 4.3? I'm not sure my first thought will be the cleanest api especially with driver clean up in 5.0.